akka-http-testkit/src/test/scala/akka/http/scaladsl/testkit/ScalatestRouteTestSpec.scala

```diff
@@ -48,6 +51,15 @@ class ScalatestRouteTestSpec extends FreeSpec with Matchers with ScalatestRouteT
       }
     }
 
+    "running on akka dispatcher threads" in Await.result(Future {
+      // https://github.com/akka/akka-http/pull/2526
+      // Check will block while waiting on the response, this might lead to starvation
+      // on the BatchingExecutor of akka's dispatcher if the blocking is not managed properly.
+      Get() ~> complete(Future(HttpResponse())) ~> check {
+        status shouldEqual OK
+      }
+    }, 5.seconds)
```

Comment:
Why would a user run their tests on the default dispatcher, isn't that a user error? Also: Isn't the test threads pretty much always managed by the test framework used?

Reply:
Yeah, I also think the use case is somewhat bogus (see #2526 and the linked sttp PR). In any case, I guess it still makes sense to try to be a good citizen and manage blocking if possible to make starvation issues less likely.

Reply:
Also, first time for me to see how easy it is to deadlock yourself with the BatchingExecutor if you don't manage blocking. `Await.result` etc do manage blocking so they already try to circumvent the problem to some degree. But of course, it's not a panacea: there are usually only so many extra threads to divert to until you might still get starved.